yarn-project/aztec-sandbox/src/sandbox.ts

```diff
@@ -65,12 +104,45 @@ export async function createSandbox(config: Partial<SandboxConfig> = {}) {
   const hdAccount = mnemonicToAccount(config.l1Mnemonic ?? MNEMONIC);
   const privKey = hdAccount.getHdKey().privateKey;
 
-  const l1Contracts = await waitThenDeploy(aztecNodeConfig.rpcUrl, hdAccount);
+  const l1Contracts = await waitThenDeploy(aztecNodeConfig, () =>
+    deployL1Contracts(aztecNodeConfig.rpcUrl, hdAccount, localAnvil, logger),
+  );
   aztecNodeConfig.publisherPrivateKey = `0x${Buffer.from(privKey!).toString('hex')}`;
-  aztecNodeConfig.rollupContract = l1Contracts.rollupAddress;
-  aztecNodeConfig.contractDeploymentEmitterContract = l1Contracts.contractDeploymentEmitterAddress;
-  aztecNodeConfig.inboxContract = l1Contracts.inboxAddress;
-  aztecNodeConfig.registryContract = l1Contracts.registryAddress;
+  aztecNodeConfig.l1Contracts.rollupAddress = l1Contracts.l1ContractAddresses.rollupAddress;
+  aztecNodeConfig.l1Contracts.contractDeploymentEmitterAddress =
+    l1Contracts.l1ContractAddresses.contractDeploymentEmitterAddress;
+  aztecNodeConfig.l1Contracts.inboxAddress = l1Contracts.l1ContractAddresses.inboxAddress;
+  aztecNodeConfig.l1Contracts.registryAddress = l1Contracts.l1ContractAddresses.registryAddress;
+
+  const node = await AztecNodeService.createAndSync(aztecNodeConfig);
+  const rpcServer = await createAztecRPCServer(node, rpcConfig);
+
+  const stop = async () => {
+    await rpcServer.stop();
+    await node.stop();
+  };
+
+  return { node, rpcServer, l1Contracts, stop };
+}
+
+/**
+ * Create and start a new Aztec Node and RPC Server. Designed for interaction with a node network.
+ * Does not start any HTTP services nor populate any initial accounts.
+ * @param config - Optional Sandbox settings.
+ */
+export async function createP2PSandbox() {
```

Comment:
Is the only difference with `createSandbox` that this one does not deploy the L1 contracts? If so, I'd go back to a single `createSandbox` function, and add an option into the `SandboxConfig` like `skipDeployL1` that controls that behavior. Otherwise, it feels like we have a lot of duplication between `createP2PSandbox` and `createSandbox`.